### DISCUSSION TITLE
Mac Mesh Router not allowing login-invalid username/password

### DISCUSSION THREAD

**Discussion by shinomen** [Ylianst/MeshCentral#6483] [Issues]
I have a server hosted locally which is in wan/lan mode.  I login to my mesh central using a FQDN that goes through a reverse proxy.  I have things setup so that when I use the FQDN it will connect to my live WAN IP address and be port forwarded to the mesh server.  On a windows machine, I have no issue logging into the mesh router.  However, I just tried the macintosh version of mesh router and it gives an invalid username/password message when trying to login to the router.  I tried to point it to the internal IP address of the mesh server and it gave a message about the certificate not matching which I told it to ignore and continue but then it gave the same invalid login message.

I do have my server configured so that you have to append the key=secret to the end of the FQDN URL so I'm not sure if that is causing any issue when trying to login from the mesh router.  I also have 2FA enabled for the normal login to the mesh central webpage.

Can someone help me figure out how I can get the macintosh mesh router to login properly?
Thanks!

And just in case you are wondering, I am manually typing the username and password and I am sure it's correct. :)

**Comment by si458** [comment]
Don't use the mac app version of meshcentralrouter!!!

It doesn't work anymore and needs fixing/upgrading at some point

To use meshcentralrouter on mac, install wine, then run the meshcentralrouter.exe with wine!

I use it here everyday with no issues!

**Reply by shinomen** [reply]
Arggggg.  :-)    Okay thanks, at least I know now.

**Reply by si458** [reply]
I will get round to having a look at it at some point as ur about the 3rd person this month to ask why the mac app wasn't working haha
the sourcecode is in its own repo if u wanted to attempt to build/fix it urself!
